scaffoldmaker/utils/vector.py

```diff
@@ -38,3 +39,34 @@ def setMagnitude(v, mag):
     '''
     scale = mag/math.sqrt(sum(c*c for c in v))
     return [ c*scale for c in v ]
+
+def getDiagonal(m):
+    '''
+    :param m: matrix
+    :return: Diagonal elements of m in a list.
+    '''
+    width, height = len(m[0]), len(m)
+
+    def diag(sx, sy):
```

Comment:
These seven lines of code seem pretty opaque to me. I haven't time to deeply analyse them but in particular I can't see how any of the later return statements are ever reached?